File: samplejava/src/main/java/com/kazakago/preferhythm/samplejava/MyPreferences.java
package com.kazakago.preferhythm.samplejava;

import com.kazakago.preferhythm.PrefClass;
import com.kazakago.preferhythm.PrefField;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

import javax.annotation.Nonnull;

/**
 * Created by tamura_k on 2017/03/08.
 */
@PrefClass
class MyPreferences {

    @PrefField
    int intPrimitive;
    @PrefField
    int intPrimitiveWithInit = 3;

    @PrefField
    long longPrimitive;
    @PrefField
    long longPrimitiveWithInit = 5L;

    @PrefField
    float floatPrimitive;
    @PrefField
    float floatPrimitiveWithInit = 5.0f;

    @PrefField
    boolean booleanPrimitive;
    @PrefField
    boolean booleanPrimitiveWithInit = true;

    @PrefField
    Integer intObject;
    @PrefField
    Integer intObjectWithInit = 9;
    @Nonnull
    @PrefField
    Integer intObjectNonNull = 12;

    @PrefField
    Long longObject;
    @PrefField
    Long longObjectWithInit = 15L;
    @Nonnull
    @PrefField
    Long longObjectNonNull = 12L;

    @PrefField
    Float floatObject;
    @PrefField
    Float floatObjectWithInit = 15.0f;
    @Nonnull
    @PrefField
    Float floatObjectNonNull = 12.0f;

    @PrefField
    Boolean booleanObject;
    @PrefField
    Boolean booleanObjectWithInit = true;
    @Nonnull
    @PrefField
    Boolean booleanObjectNonNull = true;

    @PrefField
    String stringObject;
    @PrefField
    String stringObjectWithInit = "foo";
    @Nonnull
    @PrefField
    String stringObjectNonNull = "bar";

    @PrefField
    Set<String> stringSetObject;
    @PrefField
    Set<String> stringSetObjectWithInit = new HashSet<>(Arrays.asList("foo", "bar"));
    @Nonnull
    @PrefField
    Set<String> stringSetObjectNonNull = new HashSet<>(Arrays.asList("foo", "bar"));

}

File: samplekotlin/src/main/java/com/kazakago/preferhythm/samplekotlin/MyPreferences.kt
package com.kazakago.preferhythm.samplekotlin

import com.kazakago.preferhythm.PrefClass
import com.kazakago.preferhythm.PrefField
import java.util.*

@PrefClass
class MyPreferences {

    @PrefField
    val intObject: Int? = null
    @PrefField
    val intObjectWithInit: Int? = 9
    @PrefField
    val intObjectNonNull: Int = 12

    @PrefField
    val longObject: Long? = null
    @PrefField
    val longObjectWithInit: Long? = 15L
    @PrefField
    val longObjectNonNull: Long = 12L

    @PrefField
    val floatObject: Float? = null
    @PrefField
    val floatObjectWithInit: Float? = 15.0f
    @PrefField
    val floatObjectNonNull: Float = 12.0f

    @PrefField
    val booleanObject: Boolean? = null
    @PrefField
    val booleanObjectWithInit: Boolean? = true
    @PrefField
    val booleanObjectNonNull: Boolean = true

    @PrefField
    val stringObject: String? = null
    @PrefField
    val stringObjectWithInit: String? = "foo"
    @PrefField
    val stringObjectNonNull: String = "bar"

    @PrefField
    val stringSetObject: Set<String>? = null
    @PrefField
    val stringSetObjectWithInit: Set<String>? = HashSet(Arrays.asList("foo", "bar"))
    @PrefField
    val stringSetObjectNonNull: Set<String> = HashSet(Arrays.asList("foo", "bar"))

}